components/script/dom/xrreferencespace.rs

```diff
@@ -96,21 +141,29 @@ impl XRReferenceSpace {
         let pose = self.get_unoffset_pose(base_pose);
 
         // This may change, see https://github.com/immersive-web/webxr/issues/567
-        let offset = self.transform.get().transform();
+        let offset = self.offset.transform();
         offset.post_mul(&pose)
     }
 
     /// Gets pose represented by this space
     ///
     /// Does not apply originOffset, use get_viewer_pose instead if you need it
     pub fn get_unoffset_pose(&self, base_pose: &WebVRFrameData) -> RigidTransform3D<f64> {
-        if let Some(stationary) = self.downcast::<XRStationaryReferenceSpace>() {
-            stationary.get_unoffset_pose(base_pose)
-        } else {
-            // non-subclassed XRReferenceSpaces exist, obtained via the "identity"
-            // type. These are equivalent to the viewer pose and follow the headset
-            // around
-            XRSpace::pose_to_transform(&base_pose.pose)
+        match self.ty {
+            XRReferenceSpaceType::Local => {
+                // The eye-level pose is basically whatever the headset pose was at t=0, which
+                // for most devices is (0, 0, 0)
+                RigidTransform3D::identity()
+            },
+            XRReferenceSpaceType::Local_floor => {
+                // XXXManishearth support getting floor info from stage parameters
+
+                // Assume approximate height of 2m
+                // the floor-level space is 2m below the eye-level space, which is (0, 0, 0)
+                Vector3D::new(0., -2., 0.).into()
+            },
+            XRReferenceSpaceType::Viewer => XRSpace::pose_to_transform(&base_pose.pose),
+            _ => unimplemented!(),
```

Comment:
Can we at least support `unbounded`?

Reply:
I don't have a way of testing it, though. Part of the issue here is you need a way to test if the device supports it. We currently just reject any request for bounded or unbounded spaces.

(Also, we didn't support this before, either.)